mm2src/coins/lp_coins.rs

```diff
@@ -686,6 +686,35 @@ pub async fn lp_coinfindᵃ(ctx: &MmArc, ticker: &str) -> Result<Option<MmCoinEn
     Ok(coins.get(ticker).cloned())
 }
 
+pub async fn convert_address(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    let ticker = try_s!(req["coin"].as_str().ok_or("No 'coin' field")).to_owned();
+    let coin = match lp_coinfindᵃ(&ctx, &ticker).await {
+        Ok(Some(t)) => t,
+        Ok(None) => return ERR!("No such coin: {}", ticker),
+        Err(err) => return ERR!("!lp_coinfind({}): {}", ticker, err),
+    };
+    let coin = match coin {
+        MmCoinEnum::UtxoCoin(coin) => coin,
+        _ => return ERR!("Conversion is available between UTXO's address formats only"),
+    };
+
+    let from = try_s!(req["from"].as_str().ok_or("Expect 'from' field"));
```

Comment:
Let's start to use structs as RPC methods inputs (refactoring other methods too gradually). It will be more clear what is expected if something like 
```
struct ConvertAddressInput {
    coin: &str,
    from: &str,
    to_address_format: UtxoAddressFormat,
}
```
is used.
Please check if reference type like `&str` can be used as fields. I wonder if it works, most likely not because `req` will be consumed upon `json::from_value()`, but interested to try 🙂

Reply:
To deserialize structure from json::Value a value should be moved into the json::from_value(v), because of this the structure cannot contain a borrow a string from the value caused it will be dropped. There is way to do that like following:
`let s = json::to_string(value).unwrap();
let struct: ConvertaddressInput = json::from_str(&s).unwrap()`
But I think this way uses a lot more resources than just using String instead of &str